base/idset.jl

```diff
@@ -92,8 +92,17 @@ function sizehint!(s::IdSet, newsz)
     nothing
 end
 
+function _zero!(a::Memory{<:BitInteger})
+    t = @_gc_preserve_begin a
+    p = unsafe_convert(Ptr{Cvoid}, a)
+    T = eltype(a)
+    memset(p, 0x0, (sizeof(T) * length(a)) % UInt)
```

Comment:
This seems like an odd place to put this optimization. Is it really that helpful? If so, should it be added to `fill!`? Or we could add a proper `zero!` function.

Reply:
Oh I see now this was for bootstrapping. Rest of the question still applies though,

Reply:
The `fill!` implementation we need here is from `multidimensional.jl` since it's for a `Memory`:
```julia
julia> @which fill!(m, 0)
fill!(A::AbstractArray{T}, x) where T
     @ Base multidimensional.jl:1141
 ```

Reply:
Ah. Seems to me that should be in abstractarray.jl.